File: src/main/java/com/ject/studytrip/trip/domain/model/TripReport.java
package com.ject.studytrip.trip.domain.model;

import static org.flywaydb.core.internal.util.StringUtils.hasText;

import com.ject.studytrip.global.common.entity.BaseTimeEntity;
import com.ject.studytrip.member.domain.model.Member;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.*;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder(access = AccessLevel.PRIVATE)
public class TripReport extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column private String title;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private String startDate;

    private String endDate;

    @Column(nullable = false)
    private long studyLogCount;

    @Column(nullable = false)
    private long totalFocusHours;

    @Column(nullable = false)
    private long studyDays;

    private String imageTitle;

    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    public static TripReport of(
            Member member,
            String title,
            String content,
            String startDate,
            String endDate,
            long studyLogCount,
            long totalFocusHours,
            long studyDays,
            String imageTitle) {
        return TripReport.builder()
                .title(title)
                .content(content)
                .startDate(startDate)
                .endDate(endDate)
                .studyLogCount(studyLogCount)
                .totalFocusHours(totalFocusHours)
                .studyDays(studyDays)
                .imageTitle(imageTitle)
                .imageUrl(null)
                .member(member)
                .build();
    }

    public void updateImageUrl(String imageUrl) {
        if (hasText(imageUrl)) this.imageUrl = imageUrl;
    }

    public void updateDeletedAt() {
        this.deletedAt = LocalDateTime.now();
    }

    public long getTotalFocusHours() {
        return totalFocusHours;
    }

    public String getTitle() {
        return title;
    }

    public long getStudyLogCount() {
        return studyLogCount;
    }

    public long getStudyDays() {
        return studyDays;
    }

    public String getStartDate() {
        return startDate;
    }

    public Member getMember() {
        return member;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getImageTitle() {
        return imageTitle;
    }

    public Long getId() {
        return id;
    }

    public String getEndDate() {
        return endDate;
    }

    public String getContent() {
        return content;
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/domain/model/TripReport.kt
package com.ject.studytrip.trip.domain.model

import com.ject.studytrip.global.common.entity.BaseTimeEntity
import com.ject.studytrip.member.domain.model.Member
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import org.springframework.util.StringUtils.hasText
import java.time.LocalDateTime

@Entity
class TripReport protected constructor(
    var title: String?,
    @Column(nullable = false)
    var content: String,
    @Column(nullable = false)
    var startDate: String,
    var endDate: String?,
    @Column(nullable = false)
    var studyLogCount: Long,
    @Column(nullable = false)
    var totalFocusHours: Long,
    @Column(nullable = false)
    var studyDays: Long,
    var imageTitle: String?,
    var imageUrl: String? = null,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "member_id", nullable = false)
    var member: Member,
) : BaseTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set

    companion object {
        fun of(
            member: Member,
            title: String,
            content: String,
            startDate: String,
            endDate: String?,
            studyLogCount: Long,
            totalFocusHours: Long,
            studyDays: Long,
            imageTitle: String?,
        ): TripReport = TripReport(title, content, startDate, endDate, studyLogCount, totalFocusHours, studyDays, imageTitle, null, member)
    }

    fun updateImageUrl(imageUrl: String) {
        if (hasText(imageUrl)) {
            this.imageUrl = imageUrl
        }
    }

    fun updateDeletedAt(now: LocalDateTime = LocalDateTime.now()) {
        markDeleted(now)
    }
}
